File: app/src/androidTest/java/igrek/songbook/NavigationMenuTest.java
package igrek.songbook;

import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import igrek.songbook.activity.MainActivity;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static org.hamcrest.Matchers.not;

/**
 * Instrumented test, which will execute on an Android device.
 * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
 */
@RunWith(AndroidJUnit4.class)
public class NavigationMenuTest {
	
	@Rule
	public ActivityTestRule<MainActivity> rule = new ActivityTestRule<>(MainActivity.class);
	
	@Test
	public void test_navigationMenuShows() {
		
		onView(withId(R.id.nav_view)).check(matches(not(isDisplayed())));
		
		// open nav drawer
		onView(withId(R.id.navMenuButton)).perform(click());
		
		onView(withId(R.id.nav_view)).check(matches(isDisplayed()));
	}
}

File: app/src/androidTest/java/igrek/songbook/NavigationMenuTest.kt
package igrek.songbook

import android.support.test.espresso.Espresso.onView
import android.support.test.espresso.action.ViewActions.click
import android.support.test.espresso.assertion.ViewAssertions.matches
import android.support.test.espresso.matcher.ViewMatchers.isDisplayed
import android.support.test.espresso.matcher.ViewMatchers.withId
import android.support.test.rule.ActivityTestRule
import android.support.test.runner.AndroidJUnit4
import igrek.songbook.activity.MainActivity
import org.hamcrest.Matchers.not
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

/**
 * Instrumented test, which will execute on an Android device.
 * @see [Testing documentation](http://d.android.com/tools/testing)
 */
@RunWith(AndroidJUnit4::class)
class NavigationMenuTest {

    @Rule
    @JvmField
    var rule = ActivityTestRule(MainActivity::class.java)

    @Test
    fun test_navigationMenuShows() {

        onView(withId(R.id.nav_view)).check(matches(not(isDisplayed())))

        // open nav drawer
        onView(withId(R.id.navMenuButton)).perform(click())

        onView(withId(R.id.nav_view)).check(matches(isDisplayed()))
    }
}
